### PR TITLE
Change queue size

### PR DESCRIPTION
**Pull request by MarkRivers**
Fixed logic, I think it is ready to be tested and merged.

### CODE REVIEW

**Comment by ulrikpedersen** [comment]
Starting testing.

**Comment by ulrikpedersen** [comment]
Functionality question: what happens when a plugin queue is resized during processing? As it is destroyed it seems to me that all NDArrays currently waiting in the queue are silently dropped?

**Comment by MarkRivers** [comment]
Yes, that is the design and it will be documented.

**Comment by ulrikpedersen** [comment]
I've tested it with simdetector and it appears to work. When resizing (many many times) during acquisition it seems that the queue just gets dropped and you start over with a fresh/empty queue (as @MarkRivers confirmed above). Even if the user don't change the value but just hits enter the queue is scrapped and re-created.

I also played around with making the queue huge - beyond the available memory on my PC. I used a processing plugin with all calculations enabled and then started the simdetector runnnig at full speed (~198 fps). The simDetector NDArrayPool has been created with no max limit on the amount of memory it can allocate. It took a few minutes to run out of both RAM and swap - then Linux automatically killed the process. :grinning: Surprisingly the PC did not grind to a halt while this was happening.

In summary: it worked for me even when pushing the boundaries. Documentation should state that this is potentially dangerous:

 - input queues are dropped when resizing.
 - too large input queues can cause the application to be killed or crash  (or just really slow down when swapping) when a driver or producer plugin runs out of memory.

**Comment by ulrikpedersen** [comment]
ram-test
ram-test-after

**Comment by ulrikpedersen** [comment]
Ugh. Conflicts... I bet that's in NDPluginDriver.cpp with the recent merge of #163. I think the branch need to be updated with the latest changes on master.

**Comment by MarkRivers** [comment]
I can fix that.  I did it yesterday on a local merge branch.
